community/graphdb-api/src/main/java/org/neo4j/graphdb/Relationship.java

```diff
@@ -67,9 +67,20 @@
  * when nodes and relationships are deleted, which means it's bad practice to
  * refer to them this way. Instead, use application generated ids.
  */
-public interface Relationship extends PropertyContainer
+public interface Relationship extends IdentifiablePropertyContainer
 {
     /**
+     * Returns the unique id of this relationship. Ids are garbage collected
+     * over time so they are only guaranteed to be unique during a specific time
+     * span: if the relationship is deleted, it's likely that a new relationship
+     * at some point will get the old id. This makes relationship ids brittle as
+     * public APIs.
+     *
+     * @return the id of this node
+     */
+     long getId();
```

Comment:
Each of the above points of feedback have been coded, committed, and pushed. Please let me know how the PR can be improved in any way.